os/board/common/crashdump.c

```diff
@@ -38,13 +41,26 @@
  * Definitions
  ****************************************************************************/
 #if defined(CONFIG_BOARD_RAMDUMP_UART)
-#define HANDSHAKE_STRING                "RAMDUMP"
-#define HANDSHAKE_STR_LEN_MAX           (7)
+#define HANDSHAKE_STRING		"RAMDUMP"
+#define HANDSHAKE_STR_LEN_MAX		(7)
 #endif
 
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+#define MTD_PARTNAME_LEN		15
+#define COREDUMP_BUF_SIZE		(4096)
+#endif
+
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH)
+#define MTD_PARTNAME_LEN		15
+#define MTD_1K_SIZE			(1024)
+#define MTD_FLASH_WRITE_SIZE		(32 * MTD_1K_SIZE)
+#endif
+
+
 /****************************************************************************
  * Public Function Prototypes
  ****************************************************************************/
+extern uint32_t up_dumptoflash(uint32_t offset, uint32_t *buf, uint32_t bytes);
```

Comment:
Let's add this function prototype in `arch.h` like another up APIs, up_putc, up_getc and up_puts. This is difficult to find for new chipset developer.

Reply:
done